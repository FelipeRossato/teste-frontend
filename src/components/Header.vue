<template>
  <div class="header">
    <div class="logo-wrapper">
      <a href="https://skore.io/pt">
        <img 
          src="../assets/skore.svg" 
          alt="Skore Logo"
        >
      </a> 
    </div>
    <nav class="navbar">
      <a href="/">
        <img 
          src="../assets/ICON-Home.svg" 
          alt="Home Icon"
        >
      </a> 
      <a href="#">
        <img 
          src="../assets/ICON-User.svg" 
          alt="User Icon"
        >
      </a>
      <div 
        id="toggleBtn" 
        @click="$emit('toggle')"
      >
        <img 
          :src=" mode === 'light' ? require('../assets/ICON-DarkMode.svg') : require('../assets/ICON-WhiteMoon.svg') " 
          alt="DarkMode Icon"
        >
      </div>
    </nav>
  </div>
</template>

<script>
export default {
  name: 'Header',
  emits: ['toggle'],

  props: {
    mode: {
      type: String,
      default: 'light'
    }
  }
  
}
</script>

<style scoped>
  .header {
    display: flex;
    justify-content: space-between;
    height: 100px;
  }

  .logo-wrapper img {
    width: 150px;
    object-fit: contain;
    margin: 10px 20px;
  }

  .navbar {
    display: flex;
    margin-right: 5px;
  }

  .navbar img {
    width: 40px;
    margin-top: 15px;
    margin-right: 15px;
    cursor: pointer;
  }
</style>