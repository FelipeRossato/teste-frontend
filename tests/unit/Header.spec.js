import { mount } from '@vue/test-utils'

import Header from '../../src/components/Header'

describe('Header', () => {
  const wrapper = mount(Header)

  test('Renderização inicial e snapshot', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Verifica se o botão que troca o light/dark mode está funcionando', async () => {  
    await wrapper.find('#toggleBtn').trigger('click')
    expect(wrapper.emitted().toggle).toBeTruthy()
  })
})

