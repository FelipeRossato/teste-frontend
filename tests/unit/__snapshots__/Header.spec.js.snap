// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renderização inicial e snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__props": Array [],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div
        class="header"
      >
        <div
          class="logo-wrapper"
        >
          <a
            href="https://skore.io/pt"
          >
            <img
              alt="Skore Logo"
            />
          </a>
        </div>
        <nav
          class="navbar"
        >
          <a
            href="/"
          >
            <img
              alt="Home Icon"
            />
          </a>
          <a
            href="#"
          >
            <img
              alt="User Icon"
            />
          </a>
          <div
            id="toggleBtn"
          >
            <img
              alt="DarkMode Icon"
              src=""
            />
          </div>
        </nav>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.6",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "mode": "light",
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
